src/jimp2/wireworld/z8/datamangment/Element.java

```diff
@@ -2,42 +2,61 @@
 
 
 import jimp2.wireworld.z8.wireworldlogic.CellsContainer;
+import jimp2.wireworld.z8.wireworldlogic.State;
 import jimp2.wireworld.z8.wireworldlogic.World;
 
 import java.awt.*;
 
-public abstract class Element {
-
-    protected CellsContainer cellsContainer;
+public abstract class Element extends CellsContainer{
 
     protected String name;
-
     protected Point position;
 
 
+    public Element(Point position, int width, int height) {
+        super(width, height);
+
+        this.position = position;
+    }
+
+    public Element(Point position, Dimension dimension) {
+        this(position, dimension.width, dimension.height);
+    }
+
     public boolean validateSpace(World world)
     {
+        // TODO
         return true;
     }
 
-    public void insertIntoWorld(World world)
-    {
-
+    public void insertIntoWorld(World world) {
+        insertIntoWorld(world, false);
     }
 
     public void eraseFromWorld(World world)
     {
-
+        // TODO
     }
 
-
     public String getName() {
-        // TODO implement here
-        return "";
+        return name;
     }
 
-    public void setPosition() {
-        // TODO implement here
+    public void setPosition(Point worldPosition) {
+        position = worldPosition;
     }
 
+    protected void insertIntoWorld(World world, boolean withoutEmpty) {
+        validateSpace(world);
+
+        State state;
+        for (int i = 0; i < width; i++) {
+            for (int j = 0; j < height; j++) {
+                state = cells[position.x][position.y].getState();
+                if (!withoutEmpty || state != State.EMPTY) {
+                    world.cells[i + position.x][j + position.y].setState(state);
+                }
+            }
+        }
+    }
```

Comment:
Warto byłoby to przenieść pod poprzednie insertIntoWorld, żeby było jedno pod drugim